### ISSUE TITLE
continuity of elements

### ISSUE THREAD

**Issue by BalticPinguin** [libMesh/libmesh#1472]
Hi,
I had finally some desire to look, why in my installation the adaptivity_ex3 failed (when running with hp-refinement) and stubled over the issue that I have disabled second derivatives (they are not available for infinite elements) but the  `fe->get_continuity()` (see `src/error_estimation/hp_coarsentest.C`, line 599) tells that the continuity is C_ONE and, even if d2phi is not computed, we try to access it.

Since I am not into the refinement/coarsening nor elements, I don't know how to solve it best. But I think, one should
  - add respective precompiler-flags throughout the AMR-functions or
  - ensure that the continuity is not 1 if no second derivatives are available.
Probably the first one is the more reasonable approach?

What do you think about it?

**Comment by roystgnr** [comment]
Hmm... that's actually an interesting question, but I don't like either of your solutions.  Changing AMR behavior depending on --enable-second-derivatives ought to be a no-no, that's grossly surprising behavior.

We could avoid running adaptivity_ex3 with HERMITE (the only C1 element we try it with) if second derivatives are disabled, I suppose.  That's probably the correct quick fix, if you want to take a crack at it... maybe just grep for "LIBMESH_ENABLE_SECOND_DERIVATIVES 1" in libmesh_config.h, and don't run the HERMITE cases unless the grep succeeded?

But in the long run... IMHO the HPCoarsenTest behavior here is just wrong.  We're trying to minimize H2 error when we see C1 elements, and we're doing so regardless of whether or not we're actually solving an H2 problem!  adaptivity_ex3 is H1 regardless of whether we use a more-continuous-than-necessary basis or not.

Maybe that's just an academic question, since nobody uses HERMITE elements for H1 problems in the real world, and since the HPCoarsenTest isn't a great selector to begin with?  I'd still like us to get it *right*... but I'm not sure what the right way to enable that is.  Default to continuity-based norm, but add a ```HPCoarsenTest::set_system_norm(SystemNorm sn)``` method with which users can override that?  That would let users do multiphysics with mixed spaces or dimensional norms properly too.

Okay, I think that's the right thing to do, but I don't think I want to bother with it unless/until there's real user interest.  Until then disabling the HERMITE tests when second derivatives aren't supported ought to be adequate.

**Comment by jwpeterson** [comment]
> adaptivity_ex3 is H1 regardless of whether we use a more-continuous-than-necessary basis or not.

adaptivit_ex3 is L-shaped domain, isn't that H^{1-\epsilon} or something?

Anyway @BalticPinguin I'd suggest something like the following:
```
#if !defined(LIBMESH_ENABLE_SECOND_DERIVATIVES)
  libmesh_example_requires(approx_type != "HERMITE", "--enable-second");
#endif
```
So the example can "fail gracefully" in those cases.

**Comment by roystgnr** [comment]
Still in H1, no longer in H2, IIRC?  It depends on boundary conditions and forcing function, of course, but I could swear we tweaked that one so it would be smooth enough to converge but singular enough to kill asymptotic convergence rates for any uniform refinement scheme.

Your suggested short-term fix is much better than mine.

**Comment by jwpeterson** [comment]
> Still in H1, no longer in H2, IIRC?

You're right, the example docs say it is a r^{2/3} singularity. The a priori estimates typically require u in H2 for optimal H1 convergence rates, yes? Any idea what convergence rate we actually get in ex3?

**Comment by jwpeterson** [comment]
Fixed by #1485.
